connector-packager/connector_packager/xsd_validator.py

```diff
@@ -159,4 +159,12 @@ def validate_file_specific_rules(file_to_test: ConnectorFile, path_to_file: Path
                                                  "' not an allowed value. See 'Connection Field Platform Integration' section of documentation for allowed values.")
                     return False
 
+            if 'category' in child.attrib:
+                category = child.attrib['category']
+                optional = 'optional' not in child.attrib or child.attrib['optional'] == 'true'
+                if category == 'advanced' and not optional and 'default-value' not in child.attrib:
```

Comment:
in m-c-d, the value of empty string is treated as no user input.  So if the default value is an empty string, we will still have problem. 

I propose we simple enforce the fields in Advanced tab must be optional.